File: frontend/src/types/accessControl.ts
import type { CargoUsuario } from "./index";

export type ModuleCode =
  | "AGENDA"
  | "CHAT_INTERNO"
  | "CONVERSAS"
  | "CLIENTES"
  | "VEICULOS"
  | "SERVICOS"
  | "FINANCEIRO"
  | "RELATORIOS"
  | "EQUIPE";

export interface CurrentAccess {
  modules: Record<ModuleCode, boolean>;
}

export interface CompanyModule {
  code: ModuleCode;
  name: string;
  description: string;
  enabled: boolean;
}

export interface UserModulePermissions {
  user_id: number;
  name: string;
  email: string;
  role: CargoUsuario;
  active: boolean;
  permissions: Record<ModuleCode, boolean>;
  overrides: Partial<Record<ModuleCode, boolean>>;
}

export interface AccessConfiguration {
  modules: CompanyModule[];
  users: UserModulePermissions[];
}

File: app/schemas/access_control.py
from pydantic import BaseModel

from app.models.enums import CargoUsuario


class ModuleAccessOut(BaseModel):
    code: str
    name: str
    description: str
    enabled: bool
    allowed: bool
    default_allowed: bool


class CurrentAccessOut(BaseModel):
    modules: dict[str, bool]


class CompanyModuleOut(BaseModel):
    code: str
    name: str
    description: str
    enabled: bool


class CompanyModuleUpdate(BaseModel):
    enabled: bool


class UserModulePermissionsOut(BaseModel):
    user_id: int
    name: str
    email: str
    role: CargoUsuario
    active: bool
    permissions: dict[str, bool]
    overrides: dict[str, bool]


class UserModulePermissionUpdate(BaseModel):
    allowed: bool | None


class AccessConfigurationOut(BaseModel):
    modules: list[CompanyModuleOut]
    users: list[UserModulePermissionsOut]
